Dockerfile

```diff
@@ -48,8 +47,17 @@ RUN chmod -R +x ${SALT_RUNTIME_DIR}
 
 COPY assets/sbin/* /usr/local/sbin/
 
+# Add python-ldap for ldap external_auth
+# FIXME: Downgrade to pip 22.3.1 for bug: https://github.com/saltstack/salt/issues/65025
+RUN salt-pip install "pip==22.3.1"
+RUN salt-pip install python-ldap
```

Comment:
If there is an issue with the current `pip` version, I'm fine downgrading it.

But I'm not sure if I prefer to leave the `python-ldap` package optional for users who need it rather than installing it by default.

These users can install the package as a Python extra package: https://github.com/cdalvaro/docker-salt-master?tab=readme-ov-file#python-extra-packages

You should add to the LDAP documentation that this package has to be installed using this method in order to work.

On the other hand, if we install the package by default, y prefer to do it again in the [`assets/build/install.sh`](assets/build/install.sh) script.

https://github.com/cdalvaro/docker-salt-master/blob/44c73c718b652b94b180c9214acdb9c8417e04b1/assets/build/install.sh#L54

What do you think is better?

Reply:
I like the idea of making python-ldap optional - however installing this module requires quite large build deps (gcc etc) which would have to stay installed in the image for package install at run-time.

Reply:
Yes, we need to keep the dependencies installed for the package to work. That’s why I wanted to discuss whether to make the installation of the python-ldap package optional or not.

If there had been no dependencies, we would have definitely made it optional.

Since the package shouldn’t take up too much space, I think we can install it by default.